internal/controller/controllers/agent_controller_test.go

```diff
@@ -1188,6 +1189,250 @@ var _ = Describe("agent reconcile", func() {
 
 })
 
+var _ = Describe("Approve CSRs", func() {
+	var (
+		c                     client.Client
+		hr                    *AgentReconciler
+		ctx                   = context.Background()
+		mockCtrl              *gomock.Controller
+		backEndCluster        *common.Cluster
+		hostRequest           ctrl.Request
+		agentKey              types.NamespacedName
+		hostId                strfmt.UUID
+		mockInstallerInternal *bminventory.MockInstallerInternals
+		mockClientFactory     *MockClusterKubeClientFactory
+	)
+
+	BeforeEach(func() {
+		c = fakeclient.NewClientBuilder().WithScheme(scheme.Scheme).Build()
+		mockCtrl = gomock.NewController(GinkgoT())
+		mockInstallerInternal = bminventory.NewMockInstallerInternals(mockCtrl)
+		mockClientFactory = NewMockClusterKubeClientFactory(mockCtrl)
+		hr = &AgentReconciler{
+			Client:                   c,
+			Scheme:                   scheme.Scheme,
+			Log:                      common.GetTestLog(),
+			Installer:                mockInstallerInternal,
+			APIReader:                c,
+			ClusterKubeClientFactory: mockClientFactory,
+		}
+		sId := strfmt.UUID(uuid.New().String())
+		hostId = strfmt.UUID(uuid.New().String())
+		commonHost := &common.Host{
+			Host: models.Host{
+				ID:        &hostId,
+				ClusterID: &sId,
+				Kind:      swag.String(models.HostKindAddToExistingClusterHost),
+				Inventory: common.GenerateTestDefaultInventory(),
+				Status:    swag.String(models.HostStatusAddedToExistingCluster),
+				Progress: &models.HostProgressInfo{
+					CurrentStage: models.HostStageDone,
+				},
+			},
+		}
+		backEndCluster = &common.Cluster{Cluster: models.Cluster{
+			ID: &sId,
+			Hosts: []*models.Host{
+				&commonHost.Host,
+			}}}
+		clusterDeployment := newClusterDeployment("clusterDeployment", testNamespace, getDefaultClusterDeploymentSpec("clusterDeployment-test", "test-cluster-aci", "pull-secret"))
+		Expect(c.Create(ctx, clusterDeployment)).To(BeNil())
+		mockInstallerInternal.EXPECT().GetHostByKubeKey(gomock.Any()).Return(commonHost, nil).AnyTimes()
+		agentKey = types.NamespacedName{
+			Namespace: testNamespace,
+			Name:      hostId.String(),
+		}
+		mockInstallerInternal.EXPECT().GetClusterByKubeKey(gomock.Any()).Return(backEndCluster, nil).Times(1)
+		secretName := fmt.Sprintf(adminKubeConfigStringTemplate, clusterDeployment.Name)
+		adminKubeconfigSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      secretName,
+				Namespace: clusterDeployment.Namespace,
+			},
+			Data: map[string][]byte{
+				"kubeconfig": []byte(BASIC_KUBECONFIG),
+			},
+		}
+		Expect(c.Create(ctx, adminKubeconfigSecret)).ShouldNot(HaveOccurred())
+	})
+
+	It("No matching node - No csrs", func() {
+		host := newAgent(hostId.String(), testNamespace, v1beta1.AgentSpec{ClusterDeploymentName: &v1beta1.ClusterReference{Name: "clusterDeployment", Namespace: testNamespace}})
```

Comment:
please put all the things that repeat under context and add it to before each

Reply:
OK